File: npm/tingodb/src/test/scala/io/scalajs/npm/tingodb/TingoDBTest.scala
package io.scalajs.npm.tingodb

import io.scalajs.nodejs.console
import io.scalajs.npm.mongodb.WriteOptions
import io.scalajs.npm.tingodb.DbClass._
import org.scalatest.FunSpec

import scala.scalajs.js
import scala.scalajs.js.annotation.ScalaJSDefined

/**
  * TingoDB Test
  * @author lawrence.daniels@gmail.com
  */
class TingoDBTest extends FunSpec {

  describe("TingoDB") {
    val tingoDB = TingoDB(new TingoDbOptions())

    it("supports creating a local MongoDB server") {
      // TODO use mock or embedded server
      /*
      //val db = new Db("test", new js.Object())
      val db = tingoDB.Db("test", new js.Object())
      console.log("db => ", db)

      val collection = db.collection("batch_document_insert_collection_safe")
      collection.insertMany(
        js.Array(new TestDoc(hello = "world_safe1"), new TestDoc(hello = "world_safe2")),
        new WriteOptions(w = 1),
        (err, result) => {
          assert(err == null, err)
          console.log("result => ", result)

          collection.findOne(new TestDoc(hello = "world_safe2"), (err: String, item: TestDoc) => {
            assert(err == null, err)
            assert("world_safe2" == item.hello)
          })
        })*/
    }

  }

  @ScalaJSDefined
  class TestDoc(val hello: String) extends js.Object

}

File: app/npmjs/mongo/tingodb/src/test/javascript/TingoDBTest.js
var Db = require('tingodb')({memStore: true}).Db,
    assert = require('assert'),
    fs = require("fs"),
    util = require("util");

var tingodb = require('tingodb')()
console.log(util.inspect(tingodb))

var db = new Db("", {});
// Fetch a collection to insert document into
var collection = db.collection("batch_document_insert_collection_safe");
// Insert a single document
collection.insert([{hello:'world_safe1'}, {hello:'world_safe2'}], {w:1}, function(err, result) {
  assert.equal(null, err);

  // Fetch the document
  collection.findOne({hello:'world_safe2'}, function(err, item) {
    console.info(JSON.stringify(item))
    assert.equal(null, err);
    assert.equal('world_safe2', item.hello);
  })
});